jena-arq/src/main/java/org/apache/jena/sparql/util/FmtUtils.java

```diff
@@ -535,6 +535,7 @@ private static boolean validPNameChar(char ch)
     {
         if ( Character.isLetterOrDigit(ch) ) return true ;
         if ( ch == '.' )    return true ;
+        if ( ch == ':' )    return true ;
```

Comment:
This is getting long enough that it might read better as a `switch`/`case`.

Reply:
If only `switch` accepted `Character.isLetterOrDigit` ... (this is not `match`!)

Reply:
Oh, missed that. Yeah, I was reading some Scala the other day and it made me sad to come back to ADT-less Java. 😞